libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -625,13 +653,19 @@ bool WaveClip::Paste(double t0, const WaveClip &other)
    {
       auto copy = std::make_shared<WaveClip>(*newClip.get(), factory, true);
       if (clipNeedsResampling)
+      {
+         if (stretchOtherBeforeResampling)
+            copy->ApplyStretchRatio();
          // The other clip's rate is different from ours, so resample
-          copy->Resample(mRate);
+         copy->Resample(mRate);
+      }
       if (clipNeedsNewFormat)
          // Force sample formats to match.
          copy->ConvertToSampleFormat(GetSampleFormats().Stored());
       newClip = std::move(copy);
    }
+   ApplyStretchRatio();
+   newClip->ApplyStretchRatio();
```

Comment:
The function might be more understandable if you do that stretch only when `!stretchOtherBeforeResampling`

But I understand that if `stretchOtherBeforeResampling`, then the line does nothing because the stretch ration is already unit.